src/crwimage_int.cpp

```diff
@@ -542,39 +543,30 @@ CiffComponent* CiffDirectory::doAdd(CrwDirs& crwDirs, uint16_t crwTagId) {
         if not found, create it
         set value
   */
-  CiffComponent* cc = nullptr;
   if (!crwDirs.empty()) {
     auto dir = crwDirs.top();
     crwDirs.pop();
     // Find the directory
     for (const auto& c : components_)
       if (c->tag() == dir.dir) {
-        cc = c;
-        break;
+        // Recursive call to next lower level directory
+        return c->add(crwDirs, crwTagId);
       }
-    if (!cc) {
-      // Directory doesn't exist yet, add it
-      auto m = std::make_unique<CiffDirectory>(dir.dir, dir.parent);
-      add(std::move(m));
-      cc = components_.back();
-    }
-    // Recursive call to next lower level directory
-    return cc->add(crwDirs, crwTagId);
+
+    // Directory doesn't exist yet, add it
+    auto m = std::make_unique<CiffDirectory>(dir.dir, dir.parent);
+    return add(std::move(m));
```

Comment:
This looks different than the old code. The call to `cc->add(crwDirs, crwTagId)` is gone.

Reply:
I think you need to do this (or revert back to using `cc`):
```suggestion
    auto m = std::make_unique<CiffDirectory>(dir.dir, dir.parent);
    // Recursive call to next lower level directory
    return add(std::move(m))->add(crwDirs, crwTagId);
```

Reply:
hrm interesting. I wonder why tests don't fail.

Reply:
I don't think we have good test coverage for this. Until #3137, I don't think our tests were running this code at all.